src/NBitcoin/Block.cs

```diff
@@ -264,6 +267,23 @@ public virtual void ReadWrite(BitcoinStream stream)
         {
             stream.ReadWrite(ref this.header);
             stream.ReadWrite(ref this.transactions);
+
+            if (stream.Serializing)
+                this.BlockSize = stream.Counter.WrittenBytes;
+        }
+
+        public long GetBlockSize(Network network, TransactionOptions transactionOptions = TransactionOptions.None)
```

Comment:
If this is called twice with different `TransactionOptions` that may affect the block size it will erroneously return the wrong size.

Reply:
Yes correct but if options is specified it will likely stay as a block with options.